xarray/core/dask_array_compat.py

```diff
@@ -39,7 +39,7 @@ def meta_from_array(x, ndim=None, dtype=None):
         """
         # If using x._meta, x must be a Dask Array, some libraries (e.g. zarr)
         # implement a _meta attribute that are incompatible with Dask Array._meta
-        if hasattr(x, "_meta") and isinstance(x, dask_array_type):
+        if hasattr(x, "_meta") and is_duck_dask_array(x):
```

Comment:
is `zarr` a `duck_dask_array`? If it is, we might have to switch back to `isinstance(x, dask_array_type)`, see the comment directly above

Reply:
oooh nice catch. I think we can delete this function now that our minimum dask version is 2.9

Reply:
> oooh nice catch. I think we can delete this function now that our minimum dask version is 2.9

Should I remove the entire `meta_from_array` function as part of this PR?

Reply:
Lets do that in another PR.